src/toolkit/matl_buy_policy.h

```diff
@@ -85,15 +86,27 @@ class MatlBuyPolicy : public Trader {
   /// nearest integer number of time steps from a truncated normal
   /// distribution from a mean, standard deviation, min, and max value.
   /// Note that the (s, S) policy is not currently compatible with active and
-  /// dormant buying perionds 
-  /// @param active the length of the on, actively buying period
-  /// @param dormant the length of the dormant period
+  /// Active and dormant buying perionds 
+  /// Active parameters:
+  /// @param active_dist an IntDistribution object that will be used to sample
+  /// the number of time steps in the "on" phase.  If not set, a fixed
+  /// distribution with a value of 1 will be used.
+  /// @param dormant_dist an IntDistribution object that will be used to sample
+  /// the number of time steps in the "off" phase.  If not set, a fixed
+  /// distribution with a value of 0 will be used (no dormant, always on)
+  /// @param size_dist a DoubleDistribution object that will be used to sample
+  /// the size of the request as a fraction of the available capacity at the 
+  /// current time step.  If not set, a fixed distribution with a value of 
+  /// 1.0 will be used.
   /// Note that active and dormant periods are note currently compatible with
   /// (s, S) inventory management
   /// @{
   MatlBuyPolicy& Init(Agent* manager, ResBuf<Material>* buf, std::string name);
   MatlBuyPolicy& Init(Agent* manager, ResBuf<Material>* buf, std::string name,
-                      double throughput, int active = 1, int dormant = 0);
+                      double throughput, 
+                      IntDistribution* active_dist = new FixedIntDist(1),
+                      IntDistribution* dormant_dist = new FixedIntDist(0),
+                      DoubleDistribution* size_dist = new FixedDoubleDist(1.0));
```

Comment:
This may create a memory leak if these are not deleted somewhere.  However, we can't **always** delete them since they may be passed in from somewhere else and we don't have the "authority" to delete them.

I think the best solution is to use `NULL` as the default and then instantiate these in the constructor if the value is `NULL`.  Then, you'll also want these variables to use `boost::shared_ptr<>`.  This should automatically track whether there continue to be references to the objects in question.

Reply:
But the defaults aren't overwritten until init, so the values would always be null at construction

Reply:
Oops! I keep getting `Init()` confused with the constructor... let me rephrase:

I think the best solution is to use `NULL` as the default and then instantiate these in `Init()` if the value is NULL. Then, you'll also want these variables to use boost::shared_ptr<>. This should automatically track whether there continue to be references to the objects in question.